bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -76,19 +86,18 @@ public Rule(String uid) {
      *            and the template is not resolved.
      * @param visibility visibility of rule
      */
-    public Rule(String uid, List<Trigger> triggers, //
-            List<Condition> conditions, //
-            List<Action> actions, //
-            List<ConfigDescriptionParameter> configDescriptions, //
-            Configuration configurations, String templateUID, Visibility visibility) {
-        this.uid = uid;
-        setTriggers(triggers);
-        setConditions(conditions);
-        setActions(actions);
-        setConfigurationDescriptions(configDescriptions);
-        setConfiguration(configurations);
+    public Rule(@Nullable String uid, @Nullable List<Trigger> triggers, @Nullable List<Condition> conditions,
+            @Nullable List<Action> actions, @Nullable List<ConfigDescriptionParameter> configDescriptions,
+            @Nullable Configuration configurations, @Nullable String templateUID, @Nullable Visibility visibility) {
+        this.uid = uid == null ? ID_PREFIX + UUID.randomUUID().toString() : uid;
```

Comment:
I would actually suggest to leave away the ID_PREFIX as it only makes the UID even longer without providing much benefit.

Reply:
ok